libs/langchain/langchain/vectorstores/pinecone.py

```diff
@@ -56,16 +57,28 @@ def __init__(
                 f"client should be an instance of pinecone.index.Index, "
                 f"got {type(index)}"
             )
+        if not isinstance(embedding, Embeddings):
+            warnings.warn(
+                "Passing in `embedding` as a Callable is deprecated. Please pass in an"
+                " Embeddings object instead."
+            )
         self._index = index
-        self._embedding_function = embedding_function
+        self._embedding = embedding
         self._text_key = text_key
         self._namespace = namespace
         self.distance_strategy = distance_strategy
 
-    @property
-    def embeddings(self) -> Optional[Embeddings]:
```

Comment:
should we have it return _embedding when possible?

Reply:
actually not sure what this is for, cc @hinthornw

Reply:
I think tracing related?